lib/auto-languageclient.ts

```diff
@@ -687,15 +697,41 @@ export default class AutoLanguageClient {
    * @param filePath path of a file that has changed in the project path
    * @return false => message will not be sent to the language server
    */
-  protected filterChangeWatchedFiles(filePath: string): boolean {
+  protected filterChangeWatchedFiles(_filePath: string): boolean {
```

Comment:
What's with all the _ naming on function parameter names?

Reply:
Since I enabled a couple optional diagnostics [here](https://github.com/atom/atom-languageclient/pull/213/files#diff-e5e546dd2eb0351f813d63d1b39dbc48R14), in particular, `noUnusedParameters`, otherwise TS compiler would complain. If unused parameters' name starts with an underscore, this diagnostic ignores it.

For when this could be useful, consider commit https://github.com/atom/atom-languageclient/pull/213/commits/e6856cbeddab81b3f1a66a001c4d719dae46364c -- `projectPath` is unused there, and the method is private. Hence, either the method should be `protected` to allow overriding, or the `projectPath` parameter is completely pointless and should be removed (I opted for the former). I would be much harder to catch this without `noUnusedParameters`.

Reply:
Is there any way of setting this to ignore on protected methods?  It looks like a leaky abstraction to see methods that you can override prefixing parameters with an underscore.

Reply:
Besides `//@ts-ignore`, which will turn off *all* diagnostics, no, not at the moment, as far as I know.

I can agree that it's leaking a bit of implementation detail, but I personally don't think this is a big deal. For one, overrides are free to use their own name for any variable (or not use any variables at all).

Perhaps, a better solution would be to have `@unused` JSDoc-like decorator, but as of TS 2.8 there's no such thing, and there are no plans to implement anything like this either.

Reply:
Also, notice that it's not necessarily clear whether or not a `protected` (or for that matter `public`) method can have unused parameters or not. If it's intended to be overridden, sure. If it's not intended to be overridden (for example, it's an utility method which children classes can use), then unused parameters are an indication of an error or leftover code. With the notable exception of `abstract` methods (which *can't* have a default implementation), it's impossible to tell from the code alone whether a given method is intended to be overrideable.